### DISCUSSION TITLE
How best to branch to grab a fresh copy, not including changes in my 'master' branch PR which was botched?

### DISCUSSION THREAD

**Discussion by AltF5** [winsiderss/systeminformer#1284] [Questions / Answers] [answered]
Sorry for the absolute noob GitHub question, but how best do you all 'branch' and 'revert commits' or just grab a fresh copy of SystemInformer / PH  for new Pull Request work? I cannot seem to figure this out and I'm going mad.

Scenario:
- I have a branch for SI / PH which GitHub only allows creating 1 of. - https://github.com/AltF5/processhacker
- I don't want to destroy my changes in 'master' for a Pull Request which wont be happening :-)  https://github.com/winsiderss/systeminformer/pull/1272
- I created a second branch, called "2" which is a copy of my original 'master' branch with modifications for that PR which will NOT be accepted.
- I want to either revert '2' to not be my modified 'master' (Commit rollback on '2' from it's 'master' copy)   [OR]   have '2' overwrite any changes that it obtained from 'master' and instead have it be a fresh copy from here... hope this makes sense.
- How do I do this? 

Also, going forth, do you leave your 'master' branch as an untouched, freshly mirrored copy of there, so that when you spawn new branches from 'master' it is unmodified?

Do you manage this via Github Desktop, VSCode, Visual Studio, or some sort of commandline tools connected to GH?

Thank you

**Comment by jxy-s** [comment, marked as answer]
You can rename your GitHub fork. Then change your remote in your local repo.

```
git remote remove origin
git remote add origin [your fork]
```

You can also change your upstream remote to point at the new upstream if you do that.

(Sorry on mobile right now, I'll help out more when I'm back at a computer.)

**Reply by jxy-s** [reply]
I'm unsure if this fully answers your question above. Honestly it's hard for me to follow what you're explaining/asking.

I would recommend working on branches from your fork instead of master. That will prevent you from getting into this situation.

It sounds like you created a branch with the changes you want to keep. So you can just reset the master branch to what is on your remote. Or, do an interactive rebase of your master to upstream and drop any commits that aren't upstream.

**Reply by AltF5** [reply]
Thank you, it sounds like I stupidly put my changes in my ‘master’ when master should rather  mirror whats publicly maintained (or here), used for creating other branches.

I’ll need to figure out how to connect git.exe (I assume git bash?) to my GitHub specially and then try to force overwrite everything in my ‘master’ so I can create fresh branches. 

Visual Studio and TFS felt so much easier in the past, lol!

**Reply by jxy-s** [reply]
Take the time to learn git on the command line. You'll thank me later. There is a git for Windows, you don't need bash if you don't want it.

**Reply by AltF5** [reply]
True. A pain, but probably worth the investment. 
I need to look into if git CLIs can operate within PowerShell, but it probably won’t lessen the pain since it will probably be very bash-like syntax :/

**Reply by fade2gray** [reply]
@AltF5 I don't do much git-wise, just tend to lurk in the repo sidelines and make suggestions, but it was once suggested I try [SmartGit](https://www.syntevo.com/smartgit/), free to use for single user non-commercial use.

**Reply by AltF5** [reply]
> @AltF5 I don't do much git-wise, just tend to lurk in the repo sidelines and make suggestions, but it was once suggested I try [SmartGit](https://www.syntevo.com/smartgit/), free to use for single user non-commercial use.

THANK YOU. This helped me
- Added a branch
- Local (mnu) --> Reset --> Select different commit
- Resumed dev in Visual Studio in this new branch
